vars/googleCloudFunctionDeploy.groovy

```diff
@@ -9,27 +9,36 @@ import groovy.json.JsonSlurper
- *   projectId           : Google Cloud Project ID (String, required)
- *                         - make sure the project has Jenkins worker as a principal (currently for qa and prod)
- *   gcsPath             : Google Cloud Storage path for the function archive (String, required)
- *   runtime             : Runtime for the Cloud Function (String, required)
- *   region              : Deployment region (String, required)
- *   serviceAccountEmail : Service account email for the function (String, required)
- *   environmentVariables: Map of environment variables for the function (Map, required)
- *   entryPoint          : Entry point for the function (String, optional)
- *   timeout             : Function timeout (String, optional)
- *   maxInstances        : Maximum number of function instances (Integer, optional)
- *   triggerType         : Type of trigger - 'http' or 'pubsub' (String, required)
- *   topicName           : PubSub topic name (required if triggerType is 'pubsub')
- *   generation          : Cloud Function generation - 'gen1' or 'gen2' (String, required)
- *   memory              : Memory allocation for the function (String, optional)
- *   concurrency         : Maximum number of concurrent requests (Integer, optional, gen2 only)
- *   cpu                 : CPU allocation for the function (Integer, optional, gen2 only)
+ *   functionName           : Name of the Cloud Function (String, required)
+ *   projectId              : Google Cloud Project ID (String, required)
+ *                              - make sure the project has Jenkins worker as a principal (currently for qa and prod)
+ *   runtime                : Runtime for the Cloud Function (String, required)
+ *   region                 : Deployment region (String, required)
+ *   serviceAccountEmail    : Service account email for the function (String, required)
+ *   environmentVariables   : Map of environment variables for the function (Map, optional)
+ *   entryPoint             : Entry point for the function (String, optional)
+ *   timeout                : Function timeout (String, optional)
+ *   maxInstances           : Maximum number of function instances (Integer, optional)
+ *   triggerType            : Type of trigger - 'http' or 'pubsub' (String, required)
+ *   topicName              : PubSub topic name (required if triggerType is 'pubsub')
+ *   memory                 : Memory allocation for the function (String, optional)
+ *   concurrency            : Maximum number of concurrent requests (Integer, optional)
+ *   cpu                    : CPU allocation for the function (Integer, optional)
+ *   ingress                : Ingress settings controlling what traffic can reach the function ('all' || 'internal-only' || 'internal-and-gclb', optional, default 'all')
+ *   allowUnauthenticated   : If true this will allow all callers, without checking authentication (boolean, optional, default false)
+ *   version                : function version tag (string, required)
```

Comment:
So we'll have to specify the version each time? What if we deploy version 1.0.0 and override it by forgetting to change this on next deploy? Would that prevent rolling back?

Reply:
Putting a pause on the rollback logic for now to get a docman feature out (will add rollback functionality in next update). But this is only a label so we know what code is deployed. Rollbacks will rely on checking out code using git tag that can be specified using a TAG_VERSION pipeline parameter. I that is set we'll overrride this config.version with that. But again this is only a label and doesn't actually control what code is deployed (TAG_VERSION does that)

Forgetting to bump the version is something that should be caught in reviews for the respective repositories.